lexicons/com/atproto/label/label.json

```diff
@@ -0,0 +1,23 @@
+{
+  "lexicon": 1,
+  "id": "app.bsky.label.label",
+  "defs": {
+    "main": {
+      "type": "record",
+      "description": "Metadata tag on an atproto resource (eg, repo or record)",
+      "key": "tid",
+      "record": {
+        "type": "object",
+        "required": ["src", "uri", "val", "ts"],
+        "properties": {
+          "src": {"type": "string", "maxLength": 128},
+          "uri": {"type": "string", "maxLength": 1024},
+          "cid": {"type": "string", "maxLength": 128},
+          "val": {"type": "string", "maxLength": 128},
+          "ts": {"type": "datetime"},
+          "labelUri": {"type": "string", "maxLength": 1024}
```

Comment:
I saw your note on this, hmm!!  My gut feeling is to remove `labelUri` here, and if we need another lexicon for the streaming event then that's no biggie.  In my view, the streaming event lexicons can be treated pretty much just the same as any other XRPC lexicon— the difference of responses being an asynchronous stream of values rather than a single value doesn't seem like it has a major impact on how they should be spec'ed in lexicon, at least the way I am currently thinking about it.

Also coming to mind, if we want these records to be verifiable then we need to present them as-is over the stream so that anyone can get their canonical bytes and cid.  So if it's unspec'ed whether the record is stored with `labelUri`, but over the stream it comes with `labelUri`, then the consumer wouldn't be able to figure out the exact bytes for the record.

Reply:
agreed on leaving it off here but including it over the wire 👍

Reply:
Yeah, makes sense.

For over-the-wire, do you think it would be best to create a "labelEvent" additional def here in this lexicon which has two fields: `label (com.atproto.label.label#main)` and `ref (com.atproto.repo.recordRef)` (or maybe `labelRef`)? Or, to be more similar to `#view` and `#viewDetail` in `com.atptroto.admin.moderationAction`, the "labelEvent" type could be basically the same as "label" but with the `labelUri` included. Still getting working through Lexicon schema def features and style.

Reply:
Yeah, those all feel in the ballpark for me 👍 On notifications, post, and profile views you often see something like this, which might adapt nicely here: https://github.com/bluesky-social/atproto/blob/e8c218b5c90391cd6402c43a532fc398573aa664/lexicons/app/bsky/actor/updateProfile.json#L39-L50

Reply:
I'm going to mark this resolved. I used the structure of this last `updateProfile` example, except I named the record field `label` and set the type explicitly as a ref (not `unknown`).